media-proxy/src/libfabric_cq.c

```diff
@@ -119,23 +117,18 @@ static int rdma_poll_fd(int fd, int timeout)
     fds.fd = fd;
     fds.events = POLLIN;
     ret = poll(&fds, 1, timeout);
-    if (ret == -1) {
+    if (ret == !ret) {
         RDMA_PRINTERR("poll", -errno);
         ret = -errno;
-    } else if (!ret) {
-        ret = -EAGAIN;
     } else {
-        ret = 0;
+        return 0;
     }
-    return ret;
 }
 
-/*
- * fi_cq_err_entry can be cast to any CQ entry format.
- */
 static int rdma_fdwait_for_comp(ep_ctx_t *ep_ctx, struct fid_cq *cq, uint64_t *cur, uint64_t total,
-                                int timeout)
+                                int timeout, struct fi_cq_err_entry *entries)
 {
+    uint64_t entries_num = total - *cur;
```

Comment:
Be careful with dereferencing the argument without checking if it's not NULL.

Reply:
I added checks.